src/backend/storage/buffer/bufmgr.c

```diff
@@ -815,7 +817,7 @@ ReadBuffer_common(SMgrRelation smgr, char relpersistence, ForkNumber forkNum,
 	Block		bufBlock;
 	bool		found;
 	bool		isExtend;
-	bool		isLocalBuf = SmgrIsTemp(smgr);
+	bool		isLocalBuf = SmgrIsTemp(smgr) || am_wal_redo_postgres;
```

Comment:
I think it's better to think about a special mode of running Postgres vs assuming this is an inherent property of am_wal_redo_postgres

for example when Postgres is running with wal_redo flag it also sets "Local Buffers Only" bufmgr configuration.
this way isLocalBuf = LocalBuffersOnly || SmgrIsTemp(smgr);

Reply:
Will add

Reply:
what's our flow of comment resolution? are we resolving when all of the comments are addressed?

Reply:
Comment added